pkg/tbtc/storage_test.go

```diff
@@ -0,0 +1,139 @@
+package tbtc
+
+import (
+	"github.com/keep-network/keep-common/pkg/persistence"
+	"github.com/keep-network/keep-core/pkg/internal/testutils"
+	"reflect"
+	"testing"
+)
+
+func TestSaveSigner(t *testing.T) {
+	persistenceHandle := &mockPersistenceHandle{}
+
+	walletStorage := connectWalletStorage(persistenceHandle)
+
+	signer := sampleSigner(t)
+
+	err := walletStorage.saveSigner(signer)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testutils.AssertIntsEqual(
+		t,
+		"persisted items count",
+		1,
+		len(persistenceHandle.saved),
+	)
+	testutils.AssertIntsEqual(
+		t,
+		"cached items count",
+		1,
+		len(walletStorage.wallets),
+	)
+}
+
+func TestLoadSigners(t *testing.T) {
+	signer := sampleSigner(t)
+	signerBytes, err := signer.Marshal()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	persistenceHandle := &mockPersistenceHandle{
+		saved: []persistence.DataDescriptor{
+			&mockDescriptor{
+				name:      "membership_1",
+				directory: "wallet_1",
+				content:   signerBytes,
+			},
+		},
+	}
+
+	// connectWalletStorage calls loadSigners under the hood
+	walletStorage := connectWalletStorage(persistenceHandle)
+
+	testutils.AssertIntsEqual(
+		t,
+		"cached items count",
+		1,
+		len(walletStorage.wallets),
+	)
+
+	signers := walletStorage.getSigners(signer.wallet.publicKey)
+
+	testutils.AssertIntsEqual(
+		t,
+		"wallet signers count",
+		1,
+		len(signers),
+	)
+
+	if !reflect.DeepEqual(signer, signers[0]) {
+		t.Errorf("loaded signer differs from the original one")
+	}
+}
+
+type mockPersistenceHandle struct {
+	saved []persistence.DataDescriptor
+}
+
+func (mph *mockPersistenceHandle) Save(
+	data []byte,
+	directory string,
+	name string,
+) error {
+	mph.saved = append(mph.saved, &mockDescriptor{
+		name:      name,
+		directory: directory,
+		content:   data,
+	})
+	return nil
+}
+
+func (mph *mockPersistenceHandle) Snapshot(
+	data []byte,
+	directory string,
+	name string,
+) error {
+	return nil
```

Comment:
Should we `panic` instead? This is not implemented and would be better to panic for now: no code should call this function now and before it is called, it has to be implemented. Same question about `Archive`.

Reply:
Done during 39a977d3